src/ray/core_worker/object_interface.cc

```diff
@@ -11,13 +11,17 @@ namespace ray {
 void ObjectIdsByStoreProvider(
     const std::vector<ObjectID> &object_ids,
     EnumUnorderedMap<StoreProviderType, std::unordered_set<ObjectID>> *results) {
+  // There can be a few cases here:
+  // - for task return objects, determine the store provider type for an object
+  //   from its transport type;
+  // - for other objects, just use plasma store provider.
   for (const auto &object_id : object_ids) {
-    // By default use `PLASMA` store provider.
+    // By default use `PLASMA` store provider for all the objects.
     auto type = StoreProviderType::PLASMA;
+    // Use `MEMORY` store provider for task returns objects from direct actor call.
```

Comment:
This comment just mirrors the code. It would be much more useful if it explained *why* the condition checking `IsReturnObject` is there.

Reply:
Sure, I updated to the following:

```
      // For raylet transport we always use plasma store provider, for direct actor call
       // there are a few cases:
       // - objects manually added to store by`ray.put`: for these objects they always use
       //   plasma store provider;
       // - task arguments: these objects are passed by value, and are not put into store;
       // - task return objects: these are put into memory store of the task submitter
       //   and are only used locally.
       // Thus we need to check whether this object is a task return object in additional
       // to whether it's from direct actor call before we can choose memory store provider.
```

Reply:
This looks great! One nit is that you could probably remove the last two lines because they're pretty obvious from the explanation above, but I'm not one to complain about being clear :)

Reply:
sure, will remove :)